test/test_repo.exs

```diff
@@ -91,6 +91,17 @@ end
 
 defmodule EctoSQL.TestRepo do
   use Ecto.Repo, otp_app: :ecto_sql, adapter: EctoSQL.TestAdapter
+
+  def put_dynamic_repo(name) do
+    Process.put(:current_repo, name)
+  end
+
+  defp dynamic_repo(), do: Process.get(:current_repo, __MODULE__)
+
+  def all(queryable, opts) do
+    Ecto.Repo.Queryable.all(dynamic_repo(), queryable, opts)
+  end
```

Comment:
If this patch requires the dynamic repo to work, then we need to figure out a way to not require this and explicitly call Ecto.Repo.Queryable instead of `repo.all`.

Reply:
I figured out one more direct repo call `repo.in_transaction?()` I'll try to omit it with direct call to `Ecto.Repo.Transaction.in_transaction?`

Reply:
and I found few other repo calls, replaced them all to direct call to lookup right repo_name, I'll try to refactor multiple `Keyword.get(opts, :repo_name, repo)` to put it once on the beginning

Reply:
it's better to get `Keyword.get(opts, :repo_name, repo)` with fallback in few places because of tests